### PR TITLE
Brandon presenter2

### PR DESCRIPTION
**Pull request by branflake2267**
Adding first template. This is a concept so far, see what you think. 

I'm going to rename repo to IDE-Templates.

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -0,0 +1,7 @@
+<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+	xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
```
Comment (branflake2267) [inline pom.xml:2]: 4 spaces

File: src/test/resources/com/arcbees/ide/template/presenter/__name__Module.java [hunk 2]
```diff
@@ -0,0 +1,14 @@
+#set( $symbol_pound = '#' )
+#set( $symbol_dollar = '$' )
+#set( $symbol_escape = '\' )
+package ${package};
+
+import com.gwtplatform.mvp.client.gin.AbstractPresenterModule;
+
+public class {$name}Module extends AbstractPresenterModule {
```
Comment (branflake2267) [inline src/test/resources/com/arcbees/ide/template/presenter/__name__Module.java]: ${name}

File: src/test/resources/com/arcbees/ide/template/presenter/__name__View.java [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+import com.google.gwt.uibinder.client.UiField;
+import com.google.gwt.user.client.ui.IsWidget;
+import com.google.gwt.user.client.ui.SimplePanel;
+import com.google.gwt.user.client.ui.Widget;
+import com.gwtplatform.mvp.client.ViewImpl;
+import com.gwtplatform.mvp.client.ViewWithUiHandlers;
+
+// TODO if then [ViewImpl|ViewWithUiHandlers<${name}UiHandlers>]
+public class ${name}View extends ViewWithUiHandlers<${name}UiHandlers> implements ${name}Presenter.MyView {
+    public interface Binder extends UiBinder<Widget, ${name}View> {
```
Comment (christiangoudreau) [inline src/test/resources/com/arcbees/ide/template/presenter/__name__View.java]: remove public

File: src/test/resources/com/arcbees/ide/template/presenter/__name__View.java [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+    }
+
+    @UiField
+    SimplePanel main;
+
+    @Inject
+    public ApplicationView(Binder uiBinder) {
```
Comment (christiangoudreau) [inline src/test/resources/com/arcbees/ide/template/presenter/__name__View.java]: remove public

File: src/test/resources/com/arcbees/ide/template/presenter/__name__Presenter.java [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+import com.gwtplatform.mvp.client.annotations.ContentSlot;
+import com.gwtplatform.mvp.client.annotations.ProxyStandard;
+import com.gwtplatform.mvp.client.proxy.Proxy;
+import com.gwtplatform.mvp.client.proxy.RevealContentHandler;
+import com.gwtplatform.mvp.client.HasUiHandlers;
+
+// TODO optional on uihandlers
+public class ${name}Presenter extends Presenter<${name}Presenter.MyView, ${name}Presenter.MyProxy> 
+	implements ${name}UiHandlers {
+    public interface MyView extends View, HasUiHandlers<${name}UiHandlers> {
```
Comment (christiangoudreau) [inline src/test/resources/com/arcbees/ide/template/presenter/__name__Presenter.java]: rm public

File: src/test/resources/com/arcbees/ide/template/presenter/__name__Presenter.java [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+
+    @ContentSlot
+    public static final Type<RevealContentHandler<?>> TYPE_SetMainContent = new Type<RevealContentHandler<?>>();
+
+    @ProxyStandard
+    public interface MyProxy extends Proxy<${name}Presenter> {
+    }
+
+    @Inject
+    public ApplicationPresenter(EventBus eventBus,
```
Comment (christiangoudreau) [inline src/test/resources/com/arcbees/ide/template/presenter/__name__Presenter.java]: rm public. event bus on tis own line

**Comment by christiangoudreau** [comment]
àLooks good

File: src/test/resources/com/arcbees/ide/template/presenter/__name__Presenter.java (hunk 6 above)
Comment (branflake2267) [inline src/test/resources/com/arcbees/ide/template/presenter/__name__Presenter.java]: Just a note, but i think I'll run a formatter at the end, which will inherit the project formatting. I'll change too :)

**Comment by branflake2267** [comment]
Merging for remote testing.

**Pull request merged**